processing/src/main/java/io/druid/segment/filter/BoundFilter.java

```diff
@@ -157,17 +158,25 @@ public boolean supportsBitmapIndex(BitmapIndexSelector selector)
       final BitmapIndex bitmapIndex
   )
   {
+    return Filters.bitmapsFromIndexes(getBitmapIndexIterator(boundDimFilter, bitmapIndex), bitmapIndex);
+  }
+
+  private static Iterable<Integer> getBitmapIndexIterator(
```

Comment:
I suggest to use specialized `IntIterator`. E. g. we have 100s of thousands of values in some dimensions, don't want to do boxing on this scale.